### PR TITLE
Multiple branches support - new features

### PR DESCRIPTION
**Pull request by paulovmr**
AF-1421: Allow the removal of a branch
AF-1276: Persist current branch per project of each user

Part of an ensemble:
* https://github.com/kiegroup/appformer/pull/442
* https://github.com/kiegroup/kie-wb-common/pull/2074

### CODE REVIEW

**Comment by paulovmr** [comment]
@ederign I've fixed it, it is in [this PR](https://github.com/kiegroup/appformer/pull/442). Thanks for catching it.

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by barboras7** [comment]
Hi @paulovmr,
thanks for the PR. I'll take a look at it this afternoon.

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
jenkins execute full downstream build

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/test/java/org/kie/workbench/common/screens/library/api/preferences/LibraryInternalPreferencesTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        libraryInternalPreferences.setLastBranchOpened(project,
+                                                       masterBranch);
+
+        assertEquals(masterBranch.getName(), libraryInternalPreferences.getLastBranchOpened(project).get().getName());
+    }
+
+    @Test
+    public void lastBranchOpenedOfAnotherProject() {
+        final WorkspaceProject project1 = makeProject("space", "repo1", "master");
+        final WorkspaceProject project2 = makeProject("space", "repo2", "master");
```
Comment (barboras7) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/test/java/org/kie/workbench/common/screens/library/api/preferences/LibraryInternalPreferencesTest.java]: project2 is never used; you create a project with same params in the last line of this method

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/test/java/org/kie/workbench/common/screens/library/api/preferences/LibraryInternalPreferencesTest.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+import org.guvnor.common.services.project.model.WorkspaceProject;
+import org.guvnor.structure.repositories.Branch;
+import org.guvnor.structure.repositories.Repository;
+import org.junit.Before;
+import org.junit.Test;
+import org.uberfire.spaces.Space;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.anyString;
```
Comment (barboras7) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-api/src/test/java/org/kie/workbench/common/screens/library/api/preferences/LibraryInternalPreferencesTest.java]: This import is never used

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/branch/delete/DeleteBranchPopUpScreen.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ */
+
+package org.kie.workbench.common.screens.library.client.screens.project.branch.delete;
+
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+
+import org.guvnor.common.services.project.model.WorkspaceProject;
```
Comment (barboras7) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/project/branch/delete/DeleteBranchPopUpScreen.java]: Unused import (as well as RepositoryService)

**Comment by paulovmr** [comment]
jenkins execute full downstream build

**Comment by paulovmr** [comment]
@barboras7 I fixed the issue, sent you an email with a new war, could you try again? Thanks.

**Comment by barboras7** [comment]
Thank you, @paulovmr! The issue is fixed 👍

**Pull request merged**